### PR TITLE
Made IntelliJ Inspection "Iteration over keySet() could be replaced with entrySet()" a error & Fixed the violations

### PR DESCRIPTION
**Pull request by hpandeycodeit**
Issue #6856
Make IntelliJ inspection "Iteration over keySet() could be replaced with entrySet()" a error

Old PR is messed up. Please consider this PR for review.

### CODE REVIEW

File: benchmarks/src/main/java/org/apache/druid/benchmark/FlattenJSONBenchmark.java [hunk 1]
```diff
@@ -92,7 +92,8 @@ public void prepare() throws Exception
   public Map<String, Object> baseline(final Blackhole blackhole)
   {
     Map<String, Object> parsed = flatParser.parseToMap(flatInputs.get(flatCounter));
-    for (String s : parsed.keySet()) {
+    for (Map.Entry<String, Object> entry : parsed.entrySet()) {
```
Comment (leventov) [inline benchmarks/src/main/java/org/apache/druid/benchmark/FlattenJSONBenchmark.java]: This can be just an iteration over `parsed.values()`. Similar in many other places in this PR.

File: benchmarks/src/main/java/org/apache/druid/benchmark/query/SelectBenchmark.java [hunk 2]
```diff
@@ -278,7 +278,8 @@ private SelectQuery incrementQueryPagination(SelectQuery query, SelectResultValu
     Map<String, Integer> pagingIdentifiers = prevResult.getPagingIdentifiers();
     Map<String, Integer> newPagingIdentifers = new HashMap<>();
 
-    for (String segmentId : pagingIdentifiers.keySet()) {
+    for (Map.Entry<String, Integer> entry : pagingIdentifiers.entrySet()) {
+      String segmentId = entry.getKey();
       int newOffset = pagingIdentifiers.get(segmentId) + 1;
```
Comment (leventov) [inline benchmarks/src/main/java/org/apache/druid/benchmark/query/SelectBenchmark.java]: Should be `entry.getValue() + 1`

File: benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java [hunk 3]
```diff
@@ -431,19 +431,19 @@ public void addRow(InputRow row)
     public void printStuff()
     {
       System.out.println();
-      for (String dim : dimensionMap.keySet()) {
+      for (Map<Object, Integer> dim : dimensionMap.values()) {
         System.out.println("DIMENSION " + dim + "\n============");
-        Map<Object, Integer> valueMap = dimensionMap.get(dim);
+        Map<Object, Integer> valueMap = dim;
```
Comment (leventov) [inline benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java]: Unnecessary variable, the iteration variable should be called `valueMap` itself.

File: benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java [hunk 4]
```diff
@@ -431,19 +431,19 @@ public void addRow(InputRow row)
...
-        for (Object val : valueMap.keySet()) {
+        for (Integer val : valueMap.values()) {
           valList.add((Comparable) val);
         }
 
         Collections.sort(valList);
 
         for (Comparable val : valList) {
-          System.out.println(" VAL: " + val + " CNT: " + valueMap.get(val));
+          System.out.println(" VAL: " + val + " CNT: " + val);
```
Comment (leventov) [inline benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java]: Logic is changed. Now prints `val` and `val`. Maybe easier to rewrite this code by using `new TreeMap(valueMap)` and then iterating that map, instead of moving to an ArrayList and sorting.

File: core/src/main/java/org/apache/druid/java/util/metrics/KeyedDiff.java [hunk 5]
```diff
@@ -45,8 +45,8 @@
   {
     assert xs.keySet().equals(ys.keySet());
     final Map<String, Long> zs = new HashMap<String, Long>();
-    for (String k : xs.keySet()) {
-      zs.put(k, xs.get(k) - ys.get(k));
+    for (Map.Entry<String, Long> entry : xs.entrySet()) {
+      zs.put(entry.getKey(), xs.get(entry.getKey()) - ys.get(entry.getKey()));
```
Comment (leventov) [inline core/src/main/java/org/apache/druid/java/util/metrics/KeyedDiff.java]: `xs.get(entry.getKey())` -> `entry.getValue()`.

Comment (leventov) [reply core/src/main/java/org/apache/druid/java/util/metrics/KeyedDiff.java]: @hpandeycodeit please fix similar problems in the rest of the PR, if any.

File: benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java (hunk 4 above)
Comment (hpandeycodeit) [reply benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java]: So what this function is actually doing? Will this suffice ?
`  
    for(Integer val: valueMap.values() {
          System.out.println(" VAL: " + val + " CNT: " + val);
        }
`

Comment (leventov) [reply benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java]: It sorts entries first. You can do
```java
new TreeMap<>(valueMap).forEach((val, cnt) -> System.out.println(" VAL: " + val + " CNT: " + cnt));

**Comment by hpandeycodeit** [comment]
@leventov it's failing at `Failed to handshake with host[https://127.0.0.1:8101]`. Can you try running the tests again?

File: benchmarks/src/main/java/org/apache/druid/benchmark/query/SelectBenchmark.java [hunk 6]
```diff
@@ -278,8 +278,9 @@ private SelectQuery incrementQueryPagination(SelectQuery query, SelectResultValu
     Map<String, Integer> pagingIdentifiers = prevResult.getPagingIdentifiers();
     Map<String, Integer> newPagingIdentifers = new HashMap<>();
 
-    for (String segmentId : pagingIdentifiers.keySet()) {
-      int newOffset = pagingIdentifiers.get(segmentId) + 1;
+    for (Map.Entry<String, Integer> entry : pagingIdentifiers.entrySet()) {
```
Comment (leventov) [inline benchmarks/src/main/java/org/apache/druid/benchmark/query/SelectBenchmark.java]: It could be just
```java
pagingIdentifiers.replaceAll(id, offset) -> offset + 1);
```

File: benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java [hunk 7]
```diff
@@ -431,21 +432,11 @@ public void addRow(InputRow row)
...
-      for (String dim : dimensionMap.keySet()) {
-        System.out.println("DIMENSION " + dim + "\n============");
-        Map<Object, Integer> valueMap = dimensionMap.get(dim);
-
-        List<Comparable> valList = new ArrayList<>();
-        for (Object val : valueMap.keySet()) {
-          valList.add((Comparable) val);
-        }
+      for (Map<Object, Integer> valueMap : dimensionMap.values()) {
+        System.out.println("DIMENSION " + valueMap + "\n============");
```
Comment (leventov) [inline benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java]: Now it prints `valueMap` twice. It should be `entrySet()` iteration (or better `dimensionMap.forEach()` call for readability).

File: extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java [hunk 8]
```diff
@@ -335,7 +335,8 @@ void checkSegmentsAndSubmitTasks()
     Map<Interval, String> toDropInterval = new HashMap<>(difference.entriesOnlyOnRight());
     // if some intervals are in running tasks and the versions are the same, remove it from toBuildInterval
     // if some intervals are in running tasks, but the versions are different, stop the task. 
-    for (Interval interval : runningVersion.keySet()) {
+    for (Map.Entry<Interval, String> entry : runningVersion.entrySet()) {
```
Comment (leventov) [inline extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java]: What's the purpose of this change?

File: extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java [hunk 9]
```diff
@@ -352,7 +353,8 @@ void checkSegmentsAndSubmitTasks()
       }
     }
     // drop derivative segments which interval equals the interval in toDeleteBaseSegments 
-    for (Interval interval : toDropInterval.keySet()) {
+    for (Map.Entry<Interval, String> entry : toDropInterval.entrySet()) {
```
Comment (leventov) [inline extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java]: What's the purpose of this change?

File: extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java (hunk 8 above)
Comment (hpandeycodeit) [reply extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java]: I assumed we need to replace all the occurrences of keyset() with entrySet()?

File: extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java (hunk 9 above)
Comment (hpandeycodeit) [reply extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java]: same for this one

File: extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java (hunk 8 above)
Comment (leventov) [reply extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java]: @hpandeycodeit see the description of the inspection (bold is mine):

> Reports iteration over the keySet() of a java.util.Map instance, **where the iterated keys are used to retrieve the values from the map**. Such iteration may be more efficiently replaced by iteration over the entrySet() of the map.

When values are not used in the loop, it doesn't make any sense to replace `keySet()` with `entrySet()`.

File: benchmarks/src/main/java/org/apache/druid/benchmark/query/SelectBenchmark.java (hunk 6 above)
Comment (hpandeycodeit) [reply benchmarks/src/main/java/org/apache/druid/benchmark/query/SelectBenchmark.java]: updated this

File: benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java (hunk 7 above)
Comment (hpandeycodeit) [reply benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java]: done

File: extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java (hunk 8 above)
Comment (hpandeycodeit) [reply extensions-contrib/materialized-view-maintenance/src/main/java/org/apache/druid/indexing/materializedview/MaterializedViewSupervisor.java]: reverted this and other similar change on this file

File: benchmarks/src/main/java/org/apache/druid/benchmark/query/SelectBenchmark.java (hunk 6 above)
Comment (leventov) [reply benchmarks/src/main/java/org/apache/druid/benchmark/query/SelectBenchmark.java]: `pagingIdentifiers.replaceAll((id, offset) -> offset + 1);` should replace the whole loop, not be done on every iteration.

@hpandeycodeit please read your code with fresh eye every time before committing.

File: benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java [hunk 10]
```diff
@@ -431,22 +432,8 @@ public void addRow(InputRow row)
...
-
-        Collections.sort(valList);
-
-        for (Comparable val : valList) {
-          System.out.println(" VAL: " + val + " CNT: " + valueMap.get(val));
-        }
-        System.out.println();
-      }
+      
+      dimensionMap.forEach((key, valueMap) -> new TreeMap<>(valueMap).forEach((val, cnt) -> System.out.println(" VAL: " + val + " CNT: " + cnt)));
```
Comment (leventov) [inline benchmarks/src/test/java/org/apache/druid/benchmark/BenchmarkDataGeneratorTest.java]: This line is longer than 120 cols, should be broken up

File: extensions-contrib/materialized-view-selection/src/main/java/org/apache/druid/query/materializedview/DataSourceOptimizerMonitor.java [hunk 11]
```diff
@@ -50,15 +52,15 @@ public boolean doMonitor(ServiceEmitter emitter)
...
+      for (Map.Entry<String, Long> entry : derivativesStats.entrySet()) {
+        builder.setDimension("derivative", entry.getKey());
+        emitter.emit(builder.build("/materialized/view/derivative/numSelected", derivativesStats.get(entry.getKey())));
       }
       final ServiceMetricEvent.Builder builder2 = new ServiceMetricEvent.Builder();
       builder2.setDimension("dataSource", stat.getBase());
-      for (Set<String> fields : stat.getMissFields().keySet()) {
-        builder2.setDimension("fields", fields.toString());
-        emitter.emit(builder2.build("/materialized/view/missNum", stat.getMissFields().get(fields).get()));
+      for (Map.Entry<Set<String>, AtomicLong> fields : stat.getMissFields().entrySet()) {
```
Comment (leventov) [inline extensions-contrib/materialized-view-selection/src/main/java/org/apache/druid/query/materializedview/DataSourceOptimizerMonitor.java]: Change the variable name to `e` or `entry`

File: extensions-contrib/materialized-view-selection/src/main/java/org/apache/druid/query/materializedview/DataSourceOptimizerMonitor.java [hunk 12]
```diff
@@ -50,15 +52,15 @@ public boolean doMonitor(ServiceEmitter emitter)
...
+        builder2.setDimension("fields", fields.getKey().toString());
+        emitter.emit(builder2.build("/materialized/view/missNum", stat.getMissFields().get(fields.getKey()).get()));
```
Comment (leventov) [inline extensions-contrib/materialized-view-selection/src/main/java/org/apache/druid/query/materializedview/DataSourceOptimizerMonitor.java]: Should be `e.getValue()` instead of `stat.getMissFields().get(fields.getKey())`.

@hpandeycodeit, sorry, I cannot spend so much time fixing every line in your code again and again.
